src/posix/threading.c

```diff
@@ -131,20 +131,35 @@ void sys_rwlock_unlock(sys_rwlock_t *lock)
     assert(result == 0);
 }
 
+#define SYS_THREAD_NAME_MAX 15  // thread name may be at most 15 bytes
+
+static const char thread_names[SYS_THREAD_ROLE_COUNT][SYS_THREAD_NAME_MAX + 1] = {
+    // Indexed by sys_thread_role_t. pthread names must be < 15 chars.
+    "main",          // SYS_THREAD_MAIN - root thread (not pthread!)
+    "core_thread",   // SYS_THREAD_CORE
+    "wrkr_",         // SYS_THREAD_PROACTOR (multiple)
+    "vflow_thread",  // SYS_THREAD_VFLOW
+    "lws_thread"     // SYS_THREAD_LWS_HTTP
+};
+
+static sys_atomic_t proactor_thread_count = 0;
 
 struct sys_thread_t {
     pthread_t thread;
     void *(*f)(void *);
     void *arg;
-    char  name[16];
+    void                      *proactor_context;
+    sys_thread_proactor_mode_t proactor_mode;
+    sys_thread_role_t          role;
+    char                       name[SYS_THREAD_NAME_MAX + 1];
```

Comment:
Can this be a pointer into the thread_names array ? That way you can simply point to it instead of needing to strcpy ? I am just trying to find out if what I am suggesting can be done.

Reply:
> I could, except that the thread_names array contains the name prefix in the case of the worker threads

Ah ok, I did not think about the `wrkr_` prefix

Reply:
> I could, except that the thread_names array contains the name prefix in the case of the worker threads ("wrkr_0", "wrkr_1", ... "wrkr_"N). Those names would need a separate buffer to hold the constructed name.
> 
> I can rename that array to "thread_name_prefixes" or "thread_name_formats" or some other more descriptive name. Or I can update the comment to clarify. Let me know.

Don't bother. I will mark this as Resolved